servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +453,66 @@ def test_query_jobs_submission_pagination(self):
                     page_token=response.next_page_token,
                     extensions=ExtendedQueryFields(submission=min_time)),
                 [job2])
+
+        def test_aggregtion_jobs(self):
+            labels = {
+                'aggregation-testing-unique-1': 'testing-rocks',
+                'aggregation-testing-unique-2': 'debugging-bad',
+                'aggregation-testing-unique-3': 'yeah'
+            }
+
+            # Add timestamp to distinguish from old test jobs
+            name = 'aggregation-testing' + str(datetime.datetime.now().minute)
+
+            # Create an aborted job
+            job_aborted = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            time.sleep(10)
+            self.wait_status(job_aborted, ApiStatus.RUNNING)
+            self.must_abort_job(job_aborted)
+            self.wait_status(job_aborted, ApiStatus.ABORTED)
+
+            # Create a failed job
+            job_failed = self.api_job_id(
+                self.start_job('not_a_command', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.FAILED)
+
+            # Created a succeeded job
+            job_failed = self.api_job_id(
+                self.start_job('echo SUCCEEDED', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.SUCCEEDED)
+
+            # Create jobs with specified labels above times that are large enough to
+            # be ranked top and thus appear in the Aggregation
+            for i in xrange(_LABEL_MIN_COUNT_FOR_RANK):
+                self.start_job('echo LABEL', labels=labels, name=name)
+
+            aggregationResponse = self.must_aggregate_job('HOURS_1')
+
+            for aggregation in aggregationResponse.aggregations:
```

Comment:
This feels a bit too dynamic. In general you want to form the expected response in code here, e.g. using the AggregationResponse model, then compare it. I'm not sure whether you can check equality directly on this - you may need to call `to_dict()` first.

Reply:
Accepted.